library/bignum.c

```diff
@@ -203,9 +203,15 @@ int mbedtls_mpi_copy( mbedtls_mpi *X, const mbedtls_mpi *Y )
 
     X->s = Y->s;
 
-    MBEDTLS_MPI_CHK( mbedtls_mpi_grow( X, i ) );
+    if( X->n < i )
```

Comment:
This test is repeated inside `mpi_grow`, so it looks a bit off here. To me the code would be slightly more clear if the condition was inverse, and so were the branches below, but the difference would have been negligible; I'm just sharing my thoughts.

Reply:
We have to anyways check this condition to take the ```else``` branch for the optimisation. Hence it saves double checking.